### PR TITLE
Add admin framework

### PR DESCRIPTION
**Pull request by codetheweb**
Closes https://github.com/sfbrigade/compass/issues/73.

Demo:

https://github.com/sfbrigade/compass/assets/7410405/a640ef77-13c7-46dc-bda3-c727805b2667

### CODE REVIEW

File: src/backend/auth/adapter.ts [hunk 1]
```diff
@@ -35,6 +35,15 @@ export const createPersistedAuthAdapter = (
   db: KyselyDatabaseInstance
 ): Adapter => ({
   async createUser(user) {
+    const numOfUsers = await db
+      .selectFrom("user")
+      .select((qb) => qb.fn.count("user_id").as("count"))
+      .executeTakeFirstOrThrow();
+
+    // First created user is an admin
```
Comment (amantri) [inline src/backend/auth/adapter.ts:43]: Nicely done :)

File: src/backend/auth/adapter.ts [hunk 2]
```diff
@@ -35,6 +35,15 @@ export const createPersistedAuthAdapter = (
...
+    // todo: this should be pulled from an invite or something else instead of defaulting to a para - currently devs signing in are being assigned as paras
```
Comment (amantri) [inline src/backend/auth/adapter.ts:44]: I think we will want an admin panel where admins can designate others as admins.

We will likely not allow upgrading a user to admin from emails alone - we want email invites to be shareable.

**Review by amantri — APPROVED** [review]

File: src/backend/auth/adapter.ts (hunk 2 above)
Comment (codetheweb) [reply src/backend/auth/adapter.ts:44]: > I think we will want an admin panel where admins can designate others as admins.

Yep, that can be in a future PR.

> We will likely not allow upgrading a user to admin from emails alone - we want email invites to be shareable.

Not quite sure what you mean by this?

File: src/backend/auth/adapter.ts (hunk 1 above)
Comment (tessathornberry) [reply src/backend/auth/adapter.ts:43]: This works great, Max!

**Pull request merged**